src/auspex/filters/plot.py

```diff
@@ -398,15 +332,28 @@ def final_init(self):
 
 class ManualPlotter(object):
     """Establish a figure, then give the user complete control over plot creation and data."""
-    def __init__(self,  name="", x_label='X', y_label="y", notebook=False):
-        self.plot_height  = 600
-        self.webgl        = False
+    def __init__(self,  name="", x_label='X', y_label="y"):
         self.x_label      = x_label
         self.y_label      = y_label
         self.name         = name
+        self.traces = []
 
-        self.run_in_notebook = notebook
-        if self.run_in_notebook:
-            self.plot_height = round(self.plot_height*0.5)
-        self.fig = Figure(plot_width=self.plot_height, plot_height=self.plot_height,
-                          webgl=self.webgl, x_axis_label=x_label, y_axis_label=y_label)
+    def add_trace(self, name, matplotlib_kwargs={}):
+        self.traces.append({'name': name, 'matplotlib_kwargs': matplotlib_kwargs})
+
+    def add_fit_trace(self, name):
+        self.add_trace(name, matplotlib_kwargs={'linestyle': '-', 'linewidth': 2})
+
+    def add_data_trace(self, name):
+        self.add_trace(name, matplotlib_kwargs={'linestyle': ':', 'marker': '.'})
+
+    def desc(self):
+        d =    {'plot_type': 'manual',
+                'x_label':   self.x_label,
+                'y_label':   self.y_label,
+                'traces':    self.traces
+                }
+        return d
+
+    def process_direct(self, trace_name, xdata, ydata):
```

Comment:
I see that this is not actually a new name, but I'm not in love with the name `process_direct`. Why not `push`, `append` or `send`?

Alternatively, you could expose a `dict`-like interface and make this `__setitem__` where `trace_name` becomes the key.